scipy/stats/tests/test_distributions.py

```diff
@@ -1799,6 +1799,52 @@ def test_fit_simple_non_random_data(self):
         assert_allclose(loc, 4, atol=1e-15, rtol=1e-15)
 
 
+class TestPowerlaw(object):
+    @pytest.mark.parametrize('a', [.5, 1.5, 5])
+    def test_fit_shape(self, a):
+        data = stats.powerlaw.rvs(size=1000, a=a)
+        # obtain the unpenalized log-likelihood function
+
+        def ll(args, data):
+            # to get the same behavior as `powerlaw.reduce_func`, negate the
+            # log-likelihood function
+            return -np.sum(np.log(stats.powerlaw.pdf(data, *args)), axis=0)
+
+        # compare the results of the LL
+        # a warning is raised when data is generated in some situations, so
+        # it will be supressed.
+        with np.errstate(divide='ignore'):
+            _assert_less_or_close_loglike(stats.powerlaw, data, ll)
+
+    @pytest.mark.parametrize("a", [.5, 1, 1.5])
+    @pytest.mark.parametrize("loc", [0, 1])
+    @pytest.mark.parametrize("scale", [1, 5])
+    @pytest.mark.parametrize("fa", [True, False])
+    @pytest.mark.parametrize("floc", [True, False])
+    @pytest.mark.parametrize("fscale", [True, False])
+    def test_fit_MLE_comp_optimzer_combinations(self, a, loc, scale, fa, floc,
```

Comment:
[False-False-False-5-1-1] is failing. Can you investigate the obvious things, like does the solution found by the superclass `fit` method solve the equations for a>1? If you can't figure out what's going on, can you generate stand-alone code so that we can run this example in Spyder and investigate more carefully?
One thing I'm confused about is that the LL function is supposed to be -N log(scale) when a = 1, but that is not very close to the number being returned by `ll` here.

Reply:
I have been thinking about this and am perplexed. There's 1 failure here but other failures sporadically occur when generating data. So, not an isolated incident. All of the scenarios are with a=1 and all parameters free.

**>  LL function is supposed to be -N log(scale) when a = 1,**

I checked on this and I think that it still holds. Data was generated for this case: 
`TestPowerlaw::test_fit_MLE_comp_optimzer_combination[False-False-False-1-1-1] - Exception: assert (-1.304356892110009 <= -1.888575031910272 or False)`. 
```
fitted parameters:
mle: (1, 1.00398555, 0.9947961600000002) 
opt: (0.9783093989995243, 1.0039855499999997, 0.9951438337511066)
```

The LL with no penalties is matches the issue that is raised:
`ll_mle, ll_opt = (-1.3043567831784286, -1.888290992504675)`
And the LL for when a=1, `len(data)* np.log(fscale)` (negated to match behavior of distributions):
`ll_mle_a1, ll_opt_a1 = (-1.3043567831784386, -1.2169989342644327)`

The LL whether a=1 or not is `-1.3043...`, so maybe the solution to this is to use the LL for when a=1 if maximum likelihood fit our override returned results st a=1.